src/kernels/ipywidgets-message-coordination/scriptSourceProviderFactory.node.ts

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { inject, injectable } from 'inversify';
+import { IFileSystem } from '../../platform/common/platform/types.node';
+import { IPythonExecutionFactory } from '../../platform/common/process/types.node';
+import { IConfigurationService, IHttpClient, WidgetCDNs } from '../../platform/common/types';
+import { IInterpreterService } from '../../platform/interpreter/contracts';
+import { IKernel } from '../types';
+import { CDNWidgetScriptSourceProvider } from './cdnWidgetScriptSourceProvider.node';
+import { LocalWidgetScriptSourceProvider } from './localWidgetScriptSourceProvider.node';
+import { RemoteWidgetScriptSourceProvider } from './remoteWidgetScriptSourceProvider';
+import { ILocalResourceUriConverter, IWidgetScriptSourceProvider, IWidgetScriptSourceProviderFactory } from './types';
+
+@injectable()
+export class ScriptSourceProviderFactory implements IWidgetScriptSourceProviderFactory {
+    private get configuredScriptSources(): readonly WidgetCDNs[] {
+        const settings = this.configurationSettings.getSettings(undefined);
+        return settings.widgetScriptSources;
+    }
+
+    constructor(
+        @inject(IConfigurationService) private readonly configurationSettings: IConfigurationService,
+        @inject(IFileSystem) private readonly fs: IFileSystem,
+        @inject(IInterpreterService) private readonly interpreterService: IInterpreterService,
+        @inject(IPythonExecutionFactory) private readonly factory: IPythonExecutionFactory
+    ) {}
+
+    public getProviders(kernel: IKernel, uriConverter: ILocalResourceUriConverter, httpClient: IHttpClient) {
```

Comment:
This function used to be in ipyWidgetScriptSourceProvider.ts. This is the node version.